core/capabilities/capability.go

```diff
@@ -0,0 +1,145 @@
+package capabilities
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// CapabilityType is an enum for the type of capability.
+type CapabilityType string
+
+// CapabilityType enum values.
+const (
+	CapabilityTypeTrigger CapabilityType = "trigger"
+	CapabilityTypeAction  CapabilityType = "action"
+	CapabilityTypeReport  CapabilityType = "report"
+	CapabilityTypeTarget  CapabilityType = "target"
+)
+
+// IsValid checks if the capability type is valid.
+func (c CapabilityType) IsValid() error {
+	switch c {
+	case CapabilityTypeTrigger,
+		CapabilityTypeAction,
+		CapabilityTypeReport,
+		CapabilityTypeTarget:
+		return nil
+	}
+
+	return fmt.Errorf("invalid capability type: %s", c)
+}
+
+// Validatable is an interface for validating the config and inputs of a capability.
+type Validatable interface {
```

Comment:
Note to self: this shouldn't be an interface. We should provide an implementation for the validation instead and only keep the `ExampleOutput` part of the interface (which could be moved below).